### DISCUSSION TITLE
Community Samples

### DISCUSSION THREAD

**Discussion by jamespack** [microsoft/WindowsAppSDK#3477] [Q&A] [answered]
Is there appetite for community provided samples?

**Comment by LeftTwixWand** [comment, marked as answer]
Hey, I think it might be useful:

- https://github.com/microsoft/Windows-universal-samples
- https://github.com/microsoft/WindowsAppSDK-Samples
- [customers-orders-database-sample](https://learn.microsoft.com/en-us/samples/microsoft/windows-appsample-customers-orders-database/customers-orders-database-sample)

Also, I'm working on a new version of [Microsoft Inventory sample](https://github.com/microsoft/inventorysample), built on WinUI 3 here: https://github.com/LeftTwixWand/Inventory

And you can check the official Microsoft code samples for windows: [learn.microsoft.com/en-us/samples](https://learn.microsoft.com/en-us/samples/browse/?products=windows&expanded=windows)

**Reply by jamespack** [reply]
Cool. Link them here? Or did you have a different process in mind.

[TempConverter (cpp/winrt - WinUI 3)](https://github.com/jamespack/TempConverter)

**Reply by LeftTwixWand** [reply]
I think this discussion can be used to store all the WinUI 3 / Windows App SDK samples.
And, maybe, one day we will group them up under one Awesome repository, just like this one: [awesome-react](https://github.com/enaqx/awesome-react)

**Comment by goFilmyzilla** [comment]
I found it a good discussion about the multiple information you people are sharing here[.](https://gofilmyzilla.com/) I really appreciate your efforts.

**Comment by jamespack** [comment]
[Simple MVVM Example in Cpp/WinRT](https://github.com/jamespack/winrt-cpp-mvvm)

Demonstrates getting Mica working, setting up and using MVVM pattern with ViewModel implemented navigation, binding to ViewModel properties and methods for event handling, setting the selected item from a bound menu items source and the selection indicator working (I saw a few issues filed related to this) and more.

Im so excited about the possibilities with the Windows App SDK!! 

There are a few issues that I hope get better with time. 

- The Titlebar API seems disjointed and messy. Hopefully that gets worked out because its pretty nice the flexibility you get there.
- The resize issue where content resizes slower than the window.

**Reply by LeftTwixWand** [reply]
Cool, I didn't see the C++ MVVM before

**Comment by goFilmyzilla** [comment]
Demonstrates getting Mica working, setting up and using MVVM pattern with ViewModel implemented navigation[,](https://apkcombos.net/) binding to ViewModel properties and methods for event handling.

**Comment by goFilmyzilla** [comment]
It is a good information you are sharing with the people.

**Comment by LeftTwixWand** [comment]
I think we also can add as a showcase such tools:
- [Template Studio](https://github.com/microsoft/TemplateStudio)
- [Microsoft UI XAML](https://github.com/microsoft/microsoft-ui-xaml)
- [WinUI Gallery](https://github.com/microsoft/WinUI-Gallery)
- [MVVM Toolkit Sample App](https://github.com/CommunityToolkit/MVVM-Samples)

Awesome projects:
- [DevToys](https://github.com/veler/DevToys)
- [Files](https://github.com/files-community/files)

And some legacy UWP projects, that are still cool and might be updated in the future:
- [Fluent XAML Theme Editor](https://github.com/microsoft/fluent-xaml-theme-editor)
- [VanArsdel](https://github.com/microsoft/VanArsdel)

**Reply by jamespack** [reply]
Good idea!!
